### PR TITLE
[Feature] Improve SN operations in SAGAN and SNGAN

### PR DESCRIPTION
**Pull request by LeoXing1996**
1. Allow users to control the usage of SN of embedding layers in the cBN layers
2. Support used to modify eps of SN operation
3. Modify the order of the arguments for some functions

### CODE REVIEW

File: mmgen/models/architectures/sngan_proj/modules.py [hunk 1]
```diff
@@ -39,13 +39,22 @@ class SNGANGenResBlock(nn.Module):
...
             Default to ``dict(type='BigGAN')``.
     """
 
-    _default_conv_cfg = dict(kernel_size=3, stride=1, padding=1, act_cfg=None)
+    _default_conv_cfg = dict(
+        kernel_size=3,
+        stride=1,
+        padding=1,
+        act_cfg=None,
+        with_spectral_norm=False)
```
Comment (LeoXing1996) [inline mmgen/models/architectures/sngan_proj/modules.py:57]: delete with spectral_norm

File: mmgen/models/architectures/sngan_proj/modules.py [hunk 2]
```diff
@@ -79,18 +90,24 @@ def __init__(self,
...
             self.conv_cfg.update(conv_cfg)
-        self.conv_cfg.setdefault('with_spectral_norm', with_spectral_norm)
+        with_spectral_norm = with_spectral_norm or \
+            self.conv_cfg.pop('with_spectral_norm', False)
 
         self.conv_1 = ConvModule(in_channels, hidden_channels, **self.conv_cfg)
         self.conv_2 = ConvModule(hidden_channels, out_channels,
                                  **self.conv_cfg)
+        if with_spectral_norm:
+            self.conv_1.conv = spectral_norm(self.conv_1.conv, eps=sn_eps)
```
Comment (nbei) [inline mmgen/models/architectures/sngan_proj/modules.py:100]: SNConvModule

File: mmgen/models/architectures/sngan_proj/modules.py [hunk 3]
```diff
@@ -101,7 +118,10 @@ def __init__(self,
                 stride=1,
                 padding=0,
                 act_cfg=None,
-                with_spectral_norm=with_spectral_norm)
+                with_spectral_norm=False)
+            if with_spectral_norm:
+                self.shortcut.conv = spectral_norm(
```
Comment (nbei) [inline mmgen/models/architectures/sngan_proj/modules.py:123]: SNConvModule

File: mmgen/models/architectures/sngan_proj/modules.py [hunk 4]
```diff
@@ -171,26 +191,34 @@ class SNGANDiscResBlock(nn.Module):
...
             Default to ``dict(type='BigGAN')``.
     """
 
-    _default_conv_cfg = dict(kernel_size=3, stride=1, padding=1, act_cfg=None)
+    _default_conv_cfg = dict(
+        kernel_size=3,
+        stride=1,
+        padding=1,
+        act_cfg=None,
+        with_spectral_norm=False)
```
Comment (nbei) [inline mmgen/models/architectures/sngan_proj/modules.py:211]: no need to set with_spectral_norm

File: mmgen/models/architectures/sngan_proj/generator_discriminator.py [hunk 5]
```diff
@@ -82,8 +82,15 @@ class SNGANGenerator(nn.Module):
...
             conv blocks or not. Default to False.
+        with_embedding_spectral_norm (bool, optional): Whether use spectral
+            norm for embedding layers in normalization blocks or not. If not
+            specified (set as ``None``), ``with_embedding_spectral_norm`` would
+            be set as the same value as ``with_spectral_norm``.
+            Default to None.
         norm_eps (float, optional): eps for Normalization layers (both
-            conditional and non-conditional ones). Default to 1e-4.
+            conditional and non-conditional ones). Default to `1e-4`.
+        sn_eps (float, optional): eps for Spectral Normalization operation.
```
Comment (nbei) [inline mmgen/models/architectures/sngan_proj/generator_discriminator.py:92]: spectral normalization (no need to use capital characters)

File: mmgen/models/architectures/sngan_proj/generator_discriminator.py [hunk 6]
```diff
@@ -465,6 +481,7 @@ def __init__(self,
                  blocks_cfg=dict(type='SNGANDiscResBlock'),
                  act_cfg=dict(type='ReLU'),
                  with_spectral_norm=True,
+                 sn_eps=1e-12,
```
Comment (nbei) [inline mmgen/models/architectures/sngan_proj/generator_discriminator.py:484]: `sn_eps=sn_eps`

**Pull request merged**